evm_loader/program/src/evm/opcode_table.rs

```diff
@@ -45,6 +45,9 @@ macro_rules! opcode_table {
     }
 }
 
+// TODO: Where should this go?
+pub const OPCODE_STOP: u8 = 0x00;
+
```

Comment:
We could also have the table macro emit these. So far it seems we only need this one though.

Reply:
Yes, I think we should generate the constants using the `opcode_table!` macro too.

It can be done by:
- converting `$opname:literal` to `$opname:ident`;
- using `$(opnames[$opcode as usize] = stringify!($opname);)*` instead of `$(opnames[$opcode as usize] = $opname;)*`
- and using `$(pub const $opname: u8 = $opcode;)*` to generate the constants.

Reply:
If we do this, I was considering to add another row so we can have the constants prefixed with `OPCODE_`.

Reply:
I think it is simpler to create them in a separate module and refer to them as `opcodes::STOP`

Reply:
Do you mean we should split this out into a separate module `opcodes`, and use those opcode values in `opcode_table`?

Reply:
No, I mean just adding something like this at the beginning of the macro definition:
```rust
macro_rules! opcode_table {
    ($( $opcode:literal, $opname:literal, $op:path;)*) => {
        mod opcodes {
            (pub const $opname: u8 = $opcode;)*
        }
```

Reply:
Then we end up with:

```
crate::evm::opcode_table::opcodes::STOP;
```

We could re-export them under `crate::evm::opcode`. Maybe it would be better to reorganize the code a bit. 

There are:

 * opcode values
 * opcode names (`not(target_os = "solana")`)
 * opcode implementations
 * the dispatch method (hand written jump table for `target_os = "solana"`, match statement otherwise)
 
I think all of this can be put in a single module `crate::evm::opcode`. The associations and dispatch method can be generated from a single macro invocation. 

1. What do you think about this, and
2. does that fit in this PR or should I do it in a separate one?

Reply:
I was thinking just about `use crate::opcode_table::opcodes` and then in the file we can use `opcodes::STOP`.

I suggest just adding the new constant generation part to the existing macro and avoid any renames or big code moves or refactoring in the opcode area for now. I think the names and structure are good enough as they are now too.

Reply:
While I would love to just do what you suggest, it does make me in part responsible for leaving a more messy code organization than we need. From my experience, these things get harder and harder to fix over time and so I am a bit hesitant to add to the problem, even if it does not matter much in today's version of the code. I hope you can see that point of view. 

I will take your suggestion and open a follow-up PR at least.